Clippy Lints

cholcombe973 opened the pull request:
Warning!  Many breaking changes here from clippy suggestions about best practices.

cholcombe973 committed:
Clippy lints

ChrisMacNaughton commented on src/ceph.rs:
```diff
@@ -1099,7 +1110,7 @@ impl IoCtx {
 
     /// Update tmap (trivial map)
-    pub fn rados_object_tmap_update(&self, object_name: &str, update: TmapOperation) -> RadosResult<()> {
+    pub fn rados_object_tmap_update(&self, object_name: &str, update: &TmapOperation) -> RadosResult<()> {
```
Does this need to change from pass by value to a pointer?

ChrisMacNaughton commented on src/ceph.rs:
```diff
@@ -1292,7 +1303,7 @@ impl IoCtx {
 
     // Perform a compound read operation synchronously
-    pub fn rados_perform_read_operations(&self, read_op: ReadOperation) -> RadosResult<()> {
+    pub fn rados_perform_read_operations(&self, read_op: &ReadOperation) -> RadosResult<()> {
```
Does this need to change from pass by value to a pointer?

ChrisMacNaughton commented on src/ceph.rs:
```diff
@@ -1937,7 +1939,7 @@ impl Rados {
         value: &str,
         format: Option<&str>,
-        data: Vec<*mut c_char>,
+        data: &[*mut c_char],
```
Does this need to change from pass by value to a pointer?

ChrisMacNaughton commented on src/ceph.rs:
```diff
@@ -2025,7 +2027,7 @@ impl Rados {
         value: &str,
         format: Option<&str>,
-        data: Vec<*mut c_char>,
+        data: &[*mut c_char],
```
Does this need to change from pass by value to a pointer?

ChrisMacNaughton commented on src/ceph.rs:
```diff
@@ -2112,7 +2114,7 @@ impl Rados {
         value: &str,
         format: Option<&str>,
-        data: Vec<*mut c_char>,
+        data: &[*mut c_char],
```
Does this need to change from pass by value to a pointer?

ChrisMacNaughton commented on src/ceph_client.rs:
```diff
@@ -279,7 +287,7 @@ impl CephClient {
     }
 
-    pub fn run_command(&self, command: MonCommand) -> Result<String, RadosError> {
+    pub fn run_command(&self, command: &MonCommand) -> Result<String, RadosError> {
```
Does this need to change from pass by value to a pointer?

ChrisMacNaughton commented on src/cmd.rs:
```diff
@@ -590,7 +659,7 @@ pub fn osd_crush_remove(cluster_handle: &Rados, osd_id: u64, simulate: bool) ->
 
 /// Query a ceph pool.
-pub fn osd_pool_get(cluster_handle: &Rados, pool: &str, choice: &PoolOption) -> RadosResult<String> {
+pub fn osd_pool_get(cluster_handle: &Rados, pool: &str, choice: PoolOption) -> RadosResult<String> {
```
Unlike the (many) comments about moves vs pointers, the ones in this file are the reverse?

ChrisMacNaughton reviewed (changes requested):
While I'm generally happy with passing pointers instead of moving values, I'd like to understand the reasoning for these type changes, as well as the additional changes of pointer -> move

cholcombe973 committed:
Fix unsafe pointers

cholcombe973 committed:
Rustfmt and use ? operator

Also fix some double pointers and use rados_buffer_free for
ceph heap allocated memory. Prevent mem leaks

cholcombe973 commented:
Yeah all of these changes are clippy suggestions that values weren't consumed by a function so they should be passed by reference or that enum's passed by reference would be more efficient if they were copied.  I don't know how true the copy efficiency is because i haven't benchmarked it.

Xuanwo commented:
Hi, @cholcombe973, would you be interested in working on this again?

cholcombe973 commented:
Yeah I could try a rebase and clippy again to see what happens.

On Tue, Sep 3, 2024 at 7:23 AM Xuanwo ***@***.***> wrote:

> Hi, @cholcombe973 <https://github.com/cholcombe973>, would you be
> interested in working on this again?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ceph/ceph-rust/pull/37#issuecomment-2326662187>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAXKUE23AB4IUCB7HTSCEWLZUXA73AVCNFSM6AAAAABNSILQTKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMRWGY3DEMJYG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Xuanwo commented:
> Yeah I could try a rebase and clippy again to see what happens.

Perfect, thank you so much! I'm so happy you're still here, in this community.
